File: C++/Find the closest number.cpp
class Solution {
public:
    int findClosest(int n, int k, int arr[]) {
        int ind = std::lower_bound(arr, arr + n, k) - arr;
        if (ind == 0) {
            return arr[0];
        }
        if (ind == n) {
            return arr[n - 1];
        }
        int leftElement = arr[ind - 1];
        int rightElement = arr[ind];
        int leftDiff = std::abs(leftElement - k);
        int rightDiff = std::abs(rightElement - k);
        if (leftDiff == rightDiff) {
            return rightElement;
        }
        return (leftDiff < rightDiff) ? leftElement : rightElement;
    }
};

File: JAVA/Find the closest number.java
class Solution {
    public static int findClosest(int n, int k, int[] arr) {
         int left=0;
        int right=n-1;
        while(left<=right) {
            int mid=(left+right)>>1;
            if(k<arr[mid]) {
                right=mid-1;
            }else {
                left=mid+1;
            }
        }
        if(left==arr.length)return arr[right];
        if(right<0)return arr[left];
        
        int ans=arr[left];
        int diff=Math.abs(k-ans);
        if(diff>Math.abs(k-arr[right])) ans=arr[right];
        return ans;
    }
}
        
